frontend/public/components/modals/disable-application-modal.tsx

```diff
@@ -24,14 +24,21 @@ export class DisableApplicationModal extends PromiseComponent {
     const deleteOptions = {kind: 'DeleteOptions', apiVersion: 'v1', propagationPolicy: 'Foreground'};
     const killPromises = [k8sKill(SubscriptionModel, subscription, {}, deleteOptions)]
       .concat(_.get(this.props.subscription, 'status.installedCSV') && this.state.deleteCSV
-        ? k8sKill(ClusterServiceVersionModel, {metadata: {name: subscription.status.installedCSV, namespace: subscription.metadata.namespace}} as ClusterServiceVersionKind, {}, deleteOptions)
+        ? k8sKill(ClusterServiceVersionModel, {metadata: {name: subscription.status.installedCSV, namespace: subscription.metadata.namespace}} as ClusterServiceVersionKind, {}, deleteOptions).catch(() => Promise.resolve())
         : []);
+    const handleCSC = (cscName: string) => async(cscNamespace: string) => {
+      const csc: K8sResourceKind = await k8sGet(CatalogSourceConfigModel, cscName, cscNamespace);
+      const packages = csc.spec.packages.split(',').filter(pkg => pkg !== subscription.spec.name).join(',');
+      if (packages.length === 0) {
+        await k8sKill(CatalogSourceConfigModel, csc, {}, {});
+      } else {
+        await k8sUpdate(CatalogSourceConfigModel, {...csc, spec: {...csc.spec, packages}}, csc.metadata.namespace, csc.metadata.name);
+      }
+    };
 
-    this.handlePromise(Promise.all(killPromises)).then(() => {
+    this.handlePromise(Promise.all([...killPromises, handleCSC(this.props.catalogSourceConfig)])).then(() => {
       this.props.close();
-      if (new RegExp(`/${subscription.metadata.name}(/|$)`).test(window.location.pathname)) {
-        history.push(window.location.pathname.replace(subscription.metadata.name, ''));
-      }
+      history.push(`/k8s/ns/${subscription.metadata.namespace}/${SubscriptionModel.plural}`);
```

Comment:
Let's try to keep the URL building in resource-link.jsx when possible

```suggestion
      history.push(resourceListPathFromModel(SubscriptionModel, subscription.metadata.namespace));
```